ebpf/tracer.go

```diff
@@ -154,15 +159,25 @@ func (t *Tracer) Stop() {
 }
 
 func (t *Tracer) GetActiveConnections(clientID string) (*Connections, error) {
-	latestConns, latestTime, err := t.getConnections()
+	t.bufferLock.Lock()
+	defer t.bufferLock.Unlock()
+
+	latestConns, latestTime, err := t.getConnections(t.buffer[:0])
 	if err != nil {
 		return nil, fmt.Errorf("error retrieving connections: %s", err)
 	}
 
+	// Grow or shrink buffer depending on the usage
+	if len(latestConns) >= cap(t.buffer)*2 {
+		t.buffer = make([]ConnectionStats, 0, cap(t.buffer)*2)
+	} else if len(latestConns) <= cap(t.buffer)/2 {
+		t.buffer = make([]ConnectionStats, 0, cap(t.buffer)/2)
+	}
+
 	return &Connections{Conns: t.state.Connections(clientID, latestTime, latestConns)}, nil
 }
 
-func (t *Tracer) getConnections() ([]ConnectionStats, uint64, error) {
+func (t *Tracer) getConnections(active []ConnectionStats) ([]ConnectionStats, uint64, error) {
```

Comment:
nit: adding a comment to explain why `getConnections` takes a slice of ConnectionStats as argument could help reading this function